### DISCUSSION TITLE
Mojo::File hide and unhide

### DISCUSSION THREAD

**Discussion by s1037989** [mojolicious/mojo#1699] [Ideas]
What do you think about adding a couple methods to Mojo::File to hide/unhide a file by prepending or removing a dot (.) from the beginning of the basename?  The list method respects hidden file notation.

hidden/visible could return a Mojo::File object with the hidden or visible form; or hide/unhide could actually move_to the file to a hidden or unhidden state.

This is particularly useful when writing a file to a directory being monitored by inotify.  inotify create events are triggered immediately, not when the file is finished writing; therefore it is common convention to write to a hidden file first and making it visible after the file is finished writing.

**Comment by jhthorsen** [comment]
I don't see this happening, since people might also want to add ".bak", ".tmp" or something else as well.

**Reply by s1037989** [reply]
Thanks, @jhthorsen !  Fair enough.  A role then, I assume, is the appropriate approach?  What do you think about a more generic "[Transform](https://github.com/stefanadams/Mojo-File-Role-Transform)" role?
